js/languagePickerModel.js

```diff
@@ -46,6 +56,79 @@ define([
             }
 
             this.set('_languages', languages);
+        },
+            
+        onDataLoaded: function() {
+            _.defer(function() {
+                this.locationId = Adapt.offlineStorage.get('location') || null;
+                this.restoreState();
+            }.bind(this));
+
+        },
+        
+        restoreLocation: function() {
+            _.defer(function() {
```

Comment:
I think it would be sensible to add a check here to see if the value of `this.locationId` actually exists in the course, if it doesn't `Adapt.navigateToElement` throws an error.

Although we really want to encourage people to use this new feature ONLY if each of their course languages has an identical structure, some people may decide they're OK with letting the user switch languages anyway, and although that can cause some loss of tracking data there's nothing much we can do about that - but we can prevent script errors.